### PR TITLE
More robust tests for ORTModel using decoders and use_cache=True

### PR DESCRIPTION
**Pull request by fxmarty**
As per title.

Could have helped identify https://github.com/huggingface/optimum/issues/572 https://github.com/huggingface/optimum/issues/569

### CODE REVIEW

File: optimum/utils/testing_utils.py [hunk 1]
```diff
@@ -124,9 +124,13 @@ def convert_to_hf_classes(mapping_dict):
     return hf_names_dict
 
 
-def grid_parameters(parameters: Dict[str, Iterable[Any]]) -> Iterable[Dict[str, Any]]:
+def grid_parameters(parameters: Dict[str, Iterable[Any]], return_name: bool = False) -> Iterable[Dict[str, Any]]:
     """
-    Generate an iterable over the grid of all combinations of parameters
+    Generate an iterable over the grid of all combinations of parameters, with an optional test name
```
Comment (michaelbenayoun) [inline optimum/utils/testing_utils.py:129]: nit:
```suggestion
    Generates an iterable over the grid of all combinations of parameters, with an optional test name
```

**Review by michaelbenayoun — APPROVED** [review]
LGTM!

**Review by regisss — APPROVED** [review]
Thanks @fxmarty!!

File: optimum/utils/testing_utils.py [hunk 2]
```diff
@@ -124,9 +124,13 @@ def convert_to_hf_classes(mapping_dict):
     return hf_names_dict
 
 
-def grid_parameters(parameters: Dict[str, Iterable[Any]]) -> Iterable[Dict[str, Any]]:
+def grid_parameters(parameters: Dict[str, Iterable[Any]], return_name: bool = False) -> Iterable[Dict[str, Any]]:
```
[the remark this answers is not in the document]
Comment (fxmarty) [reply optimum/utils/testing_utils.py:127]: I don't think there is, I was just being lazy to modify the older tests. Will do!

[the remark this answers is not in the document]
Comment (fxmarty) [reply optimum/utils/testing_utils.py:127]: Did the modification.

**Comment by fxmarty** [comment]
Will merge and fix tests in a later PR.

**Pull request merged**